### DISCUSSION TITLE
Do I need SUEnableDownloaderService for a sandboxed app?

### DISCUSSION THREAD

**Discussion by SwiftDevJournal** [sparkle-project/Sparkle#2270] [Q&A] [answered]
I have a sandboxed app. I released a version 0.1 without having the key `SUEnableInstallerLauncherService` in the property list. I realized my mistake when I released version 0.2 and included that key in the property list.

I recently completed a version 0.3 of my app. When I run version 0.2 of my app, check for updates, and click the Install Update button, the update downloads, but when it starts extracting the download, an alert with the following message appears:

> Update Error! An error occurred while running the updater. Please try again later.

This is the same error I had when trying to update from version 0.1 to version 0.2 because of the missing `SUEnableInstallerLauncherService` key.

The documentation for the `SUEnableDownloaderService` key says to not add the key if the app requests the `com.apple.security.network.client` entitlement. In the App Sandbox settings for the app target, I selected the Incoming Connections and Outgoing Connections checkboxes. The entitlements file has a `com.apple.security.network.server` key and an `Outgoing Connections` key, but not a `com.apple.security.network.client` key.

Do I have to add the `com.apple.security.network.client` key to the entitlements file or add the `SUEnableDownloaderService` key to the property list to get updates to install in a sandboxed app?

**Comment by zorgiepoo** [comment, marked as answer]
More detailed information on the error you hit is outputted in the console logs. It may even tell you why things are failing.

"Outgoing Connections" is the same as the `com.apple.security.network.client` key I believe. You can verify the generated entitlements of your app using `codesign -d --entitlements :- "Payload/YourApp.app"`

Also be sure to read the [sandboxing guide](https://sparkle-project.org/documentation/sandboxing/), particularly the section about the "Installer Connection & Status Services" -- it's relevant even if you don't use the XPC Services and has steps for not doing so.

**Reply by SwiftDevJournal** [reply]
You are correct that "Outgoing Connections" is the same as `com.apple.security.network.client`. So I should not have the `SUEnableDownloaderService` key in the property list. Thanks for the suggestion to use the `codesign` tool to verify that.

I have a question about the information in the "Installer Connection & Status Services" section of the sandboxing guide. When would I need to add those keys to the entitlements file? If I am using the `SUEnableInstallerLauncherService` service, do I need to add the keys in the "Installer Connection & Status Services" section of the sandboxing guide?

**Reply by zorgiepoo** [reply]
Most developers that sandbox their app need to add them. Unless you opt into using the installer connection & status XPC Services, but this is very unlikely and only should be done for very special circumstances. The documentation section explains this.

**Reply by zorgiepoo** [reply]
Also for future reference if you update Sparkle or change your app's update process in the future, make sure you [test your app](https://sparkle-project.org/documentation/#6-test-sparkle-out) before you officially publish a new version.
